lib/core/test/unit/Cardano/Wallet/DB/StateMachine.hs

```diff
@@ -1235,6 +1278,40 @@ prop_parallel db =
             cmds' = addCleanDB cmds
         prettyParallelCommands cmds =<< runParallelCommands sm' cmds'
 
+-- Controls that generators and shrinkers can run within a reasonable amount of
+-- time. We have been bitten several times already by generators which took much
+-- longer than what they should, causing timeouts in the test suite.
+validateGenerators
+    :: forall s. (Arbitrary (Wallet s), GenState s)
+    => SpecWith ()
+validateGenerators = describe "Validate generators & shrinkers" $ do
+    forM_ allGenerators $ \(name, (_frequency, gen)) -> do
+        let titleGen = "Generator for " <> name
+        it titleGen $ race (threadDelay _1s) (sanityCheckGen gen) >>= \case
+            Left{}  -> expectationFailure "Timed out."
+            Right{} -> pure ()
+
+        let titleShrink = "Shrinker for " <> name
+        it titleShrink $ do
+            cmd <- generate (resize 97 gen) -- NOTE: 97 is prime
```

Comment:
Is this to catch ``scale (`mod` n)`` ?

Reply:
Partly. I wanted to generate "big enough" data and went with `resize 100` at first, but because of the `mod` in some instances, it failed at generating big enough. Hence the prime number, which should be affected by any `mod` unless we choose specifically that number.